In [5]:
import pandas as pd
import csv
import numpy

In [6]:
column_name = "doi"

## <span style="color:black" > 1 Importation des données du Web of Science </span>

In [13]:
def read_wos(year):
    wos_per_year = []
    for year in range (2013, 2024):
        wos_df = pd.read_csv("Data/Raw/{}/wos_lyon2_{}.csv".format(str(year), str(year)), sep=",",
                 encoding='utf-8')
        wos_per_year.append(wos_df)

    full_wos = pd.concat(wos_per_year)
      
    resultat_1 = pd.DataFrame({column_name: full_wos["AR"]})                      
        
    return resultat_1

In [14]:
wos_df = read_wos(2013)

In [15]:
wos_df.shape

(3008, 1)

In [17]:
wos_df.head()

doi
0  10.1016/j.system.2013.10.003
1     10.4074/S033805991300404X
2    10.1016/j.prps.2013.09.001
3                           NaN
4         10.1684/pnv.2013.0427

In [18]:
wos_df.isnull().sum()

doi    780
dtype: int64

In [19]:
wos_df.doi.describe()

count                             2228
unique                            2228
top       10.1016/j.system.2013.10.003
freq                                 1
Name: doi, dtype: object

> **Il y a 780 lignes sans DOI et 2228 lignes avec DOI qui sont tous uniques.**

## <span style="color:black" > 2 Importation des données de Pubmed </span>

In [20]:
def read_pubmed(year):
    pubmed_per_year = []
    for year in range (2013, 2024):
        pubmed_df = pd.read_csv("Data/Raw/{}/pubmed_lyon2_{}.csv".format(str(year), str(year))
                                    ,sep=",", encoding='utf-8')
        pubmed_per_year.append(pubmed_df)
    
    full_pubmed = pd.concat(pubmed_per_year)
           
    resultat_2 = pd.DataFrame({column_name: full_pubmed["DOI"]})                      
    
    return resultat_2

In [21]:
pubmed_df = read_pubmed(2013)

In [23]:
pubmed_df.shape

(259, 1)

In [25]:
pubmed_df.head()

doi
0                           NaN
1  10.1016/j.cortex.2012.10.010
2    10.1016/j.ridd.2013.02.001
3                           NaN
4  10.1016/j.actpsy.2013.04.009

In [26]:
pubmed_df.isnull().sum()

doi    3
dtype: int64

In [27]:
pubmed_df.doi.describe()

count                              256
unique                             256
top       10.1016/j.cortex.2012.10.010
freq                                 1
Name: doi, dtype: object

> **Il y a 3 lignes sans DOI et 256 lignes avec DOI qui sont tous uniques.**

## <span style="color:black" > 3 Importation des données de HAL </span>

In [28]:
def read_hal(year):
    hal_per_year = []
    for year in range (2013, 2022):
        hal_df = pd.read_csv("Data/Raw/{}/hal_lyon2_{}.csv".format(str(year), str(year))
                                    ,sep=",", encoding='utf-8')
        hal_per_year.append(hal_df)
    
    full_hal = pd.concat(hal_per_year)
           
    resultat_3 = pd.DataFrame({column_name: full_hal["doiId_s"]})                      
    
    return resultat_3

In [29]:
hal_df = read_hal(2013)

In [37]:
hal_df.shape

(29931, 1)

In [38]:
hal_df.head()

doi
0                NaN
1  10.4000/erea.3365
2                NaN
3                NaN
4                NaN

In [39]:
hal_df.isnull().sum()

doi    22389
dtype: int64

In [41]:
hal_df.describe()

doi
count                        7542
unique                       7446
top     10.4000/books.alpara.4920
freq                            8

> **Il y a 22 389 lignes sans DOI et 7542 lignes avec DOI (dont 7446 qui sont uniques).**

## <span style="color:black" > 4 Importation des données de lens </span>

In [42]:
def read_lens(year):
    lens_per_year = []
    for year in range (2013, 2022):
        lens_df = pd.read_csv("Data/Raw/{}/lens_lyon2_{}.csv".format(str(year), str(year))
                                    ,sep=",", encoding='utf-8')
        lens_per_year.append(lens_df)
    
    full_lens = pd.concat(lens_per_year)
           
    resultat_4 = pd.DataFrame({column_name: full_lens["DOI"]})                      
    
    return resultat_4

In [43]:
lens_df  = read_lens(2013)

In [44]:
lens_df.shape

(862, 1)

In [45]:
lens_df.head()

doi
0                        NaN
1                        NaN
2                        NaN
3                        NaN
4  10.1007/s10900-013-9739-0

In [46]:
lens_df.isnull().sum()

doi    522
dtype: int64

In [47]:
lens_df.doi.describe()

count                           340
unique                          340
top       10.1007/s10900-013-9739-0
freq                              1
Name: doi, dtype: object

> **Il y a 522 lignes sans DOI et 340 lignes avec DOI qui sont tous uniques.**

## <span style="color:black" > 5 Importation des données de Wikidata </span>

In [48]:
def read_wikidata(year):
    wikidata_per_year = []
    for year in range (2013, 2022):
        wikidata_df = pd.read_csv("Data/Raw/{}/wikidata_lyon2_{}.csv".format(str(year), str(year))
                                    ,sep=",", encoding='utf-8')
        wikidata_per_year.append(wikidata_df)
    
    full_wikidata = pd.concat(wikidata_per_year)
           
    resultat_5 = pd.DataFrame({column_name: full_wikidata["doi"]})                      
    
    return resultat_5

In [49]:
wikidata_df  = read_wikidata(2013)

In [50]:
wikidata_df.shape

(362, 1)

In [51]:
wikidata_df.head()

doi
0                           NaN
1       10.4000/ARCHEOMED.10003
2            10.1037/NEU0000046
3     10.1016/J.CAD.2013.06.004
4  10.1016/J.CORTEX.2013.10.001

In [52]:
wikidata_df.isnull().sum()

doi    120
dtype: int64

In [53]:
wikidata_df.doi.describe()

count                               242
unique                              229
top       10.1080/09658211.2018.1510012
freq                                  2
Name: doi, dtype: object

> **Il y a 120 lignes sans DOI et 242 lignes avec DOI (dont 229 qui sont uniques).**

## <span style="color:black" > 6 Rassembler les 5 listes de DOI (WoS, PubMed, HAL, lens, Wikidata) et enlever les doublons </span>

In [76]:
all_lyon2 = pd.concat([wos_df, pubmed_df, hal_df, lens_df, wikidata_df])
all_lyon2.shape

(34422, 1)

In [77]:
mask_doi = all_lyon2[column_name].notna()
mask_doi

0     True
1     True
2     True
3    False
4     True
     ...  
2     True
3    False
4    False
5    False
6    False
Name: doi, Length: 34422, dtype: bool

In [78]:
doi_lyon2_doublons = all_lyon2[mask_doi]
doi_lyon2_doublons.shape

(10608, 1)

In [79]:
doi_lyon2_doublons.describe()

doi
count                       10608
unique                       9105
top     10.4000/books.alpara.4920
freq                            8

In [80]:
doi_lyon2_doublons.isnull().sum()

doi    0
dtype: int64

> **Il y a 10608 lignes avec un DOI (dont 9105 qui sont uniques). 10608 = 2228 (wos) + 256 (pubmed) + 7542 (HAL) + 340 (lens) + 242 (wikidata)**

In [81]:
doi_lyon2_doublons_minuscule = doi_lyon2_doublons[column_name].str.lower()

In [82]:
doi_lyon2_doublons_df = pd.DataFrame({column_name : doi_lyon2_doublons_minuscule}) 

In [83]:
doi_lyon2_final = doi_lyon2_doublons_df.drop_duplicates()

In [84]:
doi_lyon2_final.reset_index(drop=True, inplace=True)
doi_lyon2_final.head()

doi
0  10.1016/j.system.2013.10.003
1     10.4074/s033805991300404x
2    10.1016/j.prps.2013.09.001
3         10.1684/pnv.2013.0427
4  10.1080/13554794.2012.713491

In [85]:
doi_lyon2_final.shape

(8985, 1)

In [86]:
doi_lyon2_final.describe()

doi
count                           8985
unique                          8985
top     10.1016/j.system.2013.10.003
freq                               1

> **Une fois tous les doi mis en minuscule et que les doublons sont retirés, on obtient un total de 8985 uniques doi**

In [87]:
doi_lyon2_final.to_csv("Data\Outputs\doi_lyon2.csv", index=False)

In [88]:
doi_lyon2_final.to_excel("Data\Outputs\doi_lyon2.xlsx", index = False)

In [ ]:
## J'ai ajouté à la main dans excel des colonnes pour les identifiants structure dans HAL et le code thèse

In [89]:
doi_lyon2_final_2 = pd.read_excel('Data\Outputs\doi_lyon2.xlsx')

In [91]:
doi_lyon2_final_2.shape

(8985, 6)

In [93]:
doi_lyon2_final_2.to_csv("Data\Outputs\doi_lyon2.csv", index=False, sep = ';')>